src/storage/backend/backend_impl.rs

```diff
@@ -66,10 +69,23 @@ impl BackendBuilder {
                     .endpoint(endpoint_url)
                     .access_key_id(access_key_id)
                     .secret_access_key(secret_access_key)
-                    .region("auto")
                     .bucket(bucket_name);
 
-                Operator::new(builder)?.layer(layer).finish()
+                // Auto detect region
+                let region = match S3::detect_region(endpoint_url, bucket_name).await {
+                    Some(region) => region,
+                    None => "auto".to_owned(),
+                };
+                builder.region(region.as_str());
+
+                // For aws s3 issue: https://repost.aws/questions/QU_F-UC6-fSdOYzp-gZSDTvQ/receiving-s3-503-slow-down-responses
+                // 3,500 PUT/COPY/POST/DELETE or 5,500 GET/HEAD requests per second per prefix in a bucket
+                let conncurrency_layer = ConcurrentLimitLayer::new(3500);
```

Comment:
It would be better add a config item for this concurrency setting

Reply:
ok, I will updated it with optional region config.